xbmc/platform/linux/input/LibInputKeyboard.cpp

```diff
@@ -148,39 +149,75 @@ static const std::map<xkb_keysym_t, XBMCKey> xkbMap =
   { XKB_KEY_XF86AudioRandomPlay, XBMCK_SHUFFLE }
   // XBMCK_FASTFORWARD clashes with XBMCK_MEDIA_FASTFORWARD
 };
+
+static void xkbLogger(xkb_context* context,
+                      xkb_log_level priority,
+                      const char* format,
+                      va_list args)
+{
+  const std::string message = StringUtils::FormatV(format, args);
+  auto logLevel = LOGDEBUG;
+  switch (priority)
+  {
+    case XKB_LOG_LEVEL_INFO:
+      logLevel = LOGINFO;
+      break;
+    case XKB_LOG_LEVEL_ERROR:
+      logLevel = LOGERROR;
+      break;
+    case XKB_LOG_LEVEL_WARNING:
+      logLevel = LOGWARNING;
+      break;
+    case XKB_LOG_LEVEL_DEBUG:
+      logLevel = LOGDEBUG;
+      break;
+    default:
+      break;
+  };
+  CLog::LogF(logLevel, "{}", message);
```

Comment:
Probably better to use `Log` and prefix the message with `[xkb]` or something.

Have you seen anything logged here? Is it actually useful?

Reply:
Honestly the logger is only useful for the compose stuff since it points out the compose file that is under usage (or if it fails to detect any). Right now (without composition support) it's only useful to confirm the keyboard layout and model (adds some garbage in debug mode on startup):

```
2023-10-31 15:04:21.337 T:780     debug <general>: xkbLogger: Compiling from RMLVO: rules 'evdev', model 'pc105', layout 'pt', variant '(null)', options '(null)'
2023-10-31 15:04:21.346 T:780     debug <general>: xkbLogger: Compiling from KcCGST: keycodes 'evdev+aliases(qwerty)', types 'complete', compat 'complete', symbols 'pc+pt+inet(evdev)'
2023-10-31 15:04:21.346 T:780     debug <general>: xkbLogger: Compiling xkb_keycodes "(unnamed)"
2023-10-31 15:04:21.352 T:780     debug <general>: xkbLogger: Compiling xkb_types "(unnamed)"
2023-10-31 15:04:21.372 T:780     debug <general>: xkbLogger: Compiling xkb_compatibility "(unnamed)"
2023-10-31 15:04:21.381 T:780     debug <general>: xkbLogger: The "group" statement in compat is unsupported; Ignored
2023-10-31 15:04:21.383 T:780     debug <general>: xkbLogger: The "allowExplicit" field in indicator statements is unsupported; Ignored
2023-10-31 15:04:21.388 T:780     debug <general>: xkbLogger: The "indicatorDrivesKeyboard" field in indicator statements is unsupported; Ignored
2023-10-31 15:04:21.392 T:780     debug <general>: xkbLogger: The "allowExplicit" field in indicator statements is unsupported; Ignored
2023-10-31 15:04:21.398 T:780     debug <general>: xkbLogger: Indicator name "Shift Lock" was not declared in the keycodes section; Adding new indicator
2023-10-31 15:04:21.398 T:780     debug <general>: xkbLogger: Indicator name "Group 2" was not declared in the keycodes section; Adding new indicator
2023-10-31 15:04:21.398 T:780     debug <general>: xkbLogger: Indicator name "Mouse Keys" was not declared in the keycodes section; Adding new indicator
2023-10-31 15:04:21.398 T:780     debug <general>: xkbLogger: Compiling xkb_symbols "(unnamed)"
```

For key composition:

```
2023-10-31 15:04:21.461 T:780     debug <general>: xkbLogger: created compose table from locale C with path /usr/share/X11/locale/en_US.UTF-8/Compose
```

I can do the `[xkb]` prefix if you prefer

Reply:
done